### PR TITLE
Updated report generation and fixed multiple bugs

### PR DESCRIPTION
**Pull request by artem-smotrakov**
Here is a list of main updates:

1. Updated `MultipleSecurityRatingsCalculator` to catch exceptions.
1. Updated `MarkdownReporter` to sort projects by number of stars.
1. Fixed multiple bugs (please the commit messages for details).
1. Updated tests.

### CODE REVIEW

File: src/main/java/com/sap/sgs/phosphor/fosstars/data/github/UsesOwaspDependencyCheck.java [hunk 1]
```diff
@@ -63,23 +63,29 @@ private boolean checkMaven(GHRepository repository) throws IOException {
...
+    if (content == null || !content.isFile()) {
       return false;
     }
+
     Model model = readModel(content);
-    for (Plugin plugin : model.getBuild().getPlugins()) {
-      if (isDependencyCheck(plugin)) {
-        return true;
+
+    if (model.getBuild() != null) {
```
Comment (sourabhsparkala) [inline src/main/java/com/sap/sgs/phosphor/fosstars/data/github/UsesOwaspDependencyCheck.java]: model.getBuild() maybe extract this to a variable?

File: src/main/java/com/sap/sgs/phosphor/fosstars/data/github/UsesOwaspDependencyCheck.java [hunk 2]
```diff
@@ -63,23 +63,29 @@ private boolean checkMaven(GHRepository repository) throws IOException {
...
+        if (isDependencyCheck(plugin)) {
+          return true;
+        }
       }
     }
-    for (ReportPlugin plugin : model.getReporting().getPlugins()) {
-      if (isDependencyCheck(plugin)) {
-        return true;
+
+    if (model.getReporting() != null) {
```
Comment (sourabhsparkala) [inline src/main/java/com/sap/sgs/phosphor/fosstars/data/github/UsesOwaspDependencyCheck.java]: model.getReporting() maybe extract this to a variable?

File: src/main/java/com/sap/sgs/phosphor/fosstars/tool/github/MarkdownReporter.java [hunk 3]
```diff
@@ -187,5 +207,52 @@ private static String scoreOf(GitHubProject project) {
...
 
+  /**
+   * Figures out how many stars a project has.
+   *
+   * @param project The project.
+   * @return A number of stars or {@link #UNKNOWN_NUMBER_OF_STARS}
+   *         if the number of stars is unknown.
+   */
+  private static int starsOf(GitHubProject project) {
+    if (!project.ratingValue().isPresent()) {
```
Comment (sourabhsparkala) [inline src/main/java/com/sap/sgs/phosphor/fosstars/tool/github/MarkdownReporter.java]: project.getratingValue() maybe extract this to a variable?

**Comment by artem-smotrakov** [comment]
> Overall Looks good 
> Minor comments

Thanks for the review @sourabhsparkala ! Your suggestions are totally valid. I've checked the methods, they just return values without doing any heavy-lifting (it's not always the case, for example, `GHCommit.getAuthor()` tries to resolve a user). If you don't mind I'd prefer to keep it as-is to make the code a bit shorter.

**Pull request merged**